### PR TITLE
fix: XhrPlugin cleans cache on every record

### LINKED ISSUE #452: [Bug]: XhrPlugin leaks memory profusely
### Which web client version did you detect this bug with?

v1.13.7

### What environment (build systems, module system, and framework) did you detect this bug with?

TypeScript v4.4.4, Webpack v4.42.0

### Is your web application a single page application (SPA) or multi page application (MPA)?

SPA

### Please provide your web client configuration

 {
      guestRoleArn: "arn:aws:iam::099722809053:role/orion-qa-migrate-RUMAppMonitorIAMRole-PHTR2VRVOF8X",
      identityPoolId:  "***",
      sessionSampleRate: 1,
      endpoint: "https://dataplane.rum.us-east-1.amazonaws.com",
      telemetries: ['performance', 'errors', 'http'],
      allowCookies: true,
      enableXRay: false,
 }

### Please describe the bug/issue

XhrPlugin seems to be installed at initialization.  It wraps XMLHttpRequest.  In XhrPlugin.js, all requests are put into a Map called xhrMap:
https://github.com/aws-observability/aws-rum-web/blob/main/src/plugins/event-plugins/XhrPlugin.ts#L98

This map uses each request as a key, and seems to use it to keep the request around through its lifetime in order to register events to RUM.  However, these entries are NEVER removed.  Furthermore, after they receive their responses, the response lives on in the request, which can cause **extreme** leakage for large requests.  

This totally prevents us from using it at all.

### PR DESCRIPTION
**Pull request by williazz**
closes #452 

XhrPlugin is leaking memory because the internal map is never cleaned. This is causing the web client to be unusable for customers with a high volume of xhr requests. 

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/plugins/event-plugins/__tests__/XhrPlugin.test.ts [hunk 1]
```diff
@@ -65,6 +65,7 @@ describe('XhrPlugin tests', () => {
                 statusText: 'OK'
             }
         });
+        expect(plugin.cacheSize).toEqual(0);
```
Comment (williazz) [inline src/plugins/event-plugins/__tests__/XhrPlugin.test.ts:68]: Note: I check this on every test so that there are no edge cases where the cache size net increases

**Comment by ps863** [comment]
question: is this cleanup required in any of the other plugins if we aren't doing it already?

**Review by ps863 — APPROVED** [review]

File: src/plugins/event-plugins/XhrPlugin.ts [hunk 2]
```diff
@@ -207,19 +212,25 @@ export class XhrPlugin extends MonkeyPatched<XMLHttpRequest, 'send' | 'open'> {
...
+            this.handleXhrDetailsOnError(
+                xhrDetails,
+                xhr,
+                'XMLHttpRequest abort'
+            );
+        }
     };
 
     private handleXhrTimeoutEvent = (e: Event) => {
         const xhr: XMLHttpRequest = e.target as XMLHttpRequest;
```
[the remark this answers is not in the document]
Comment (williazz) [reply src/plugins/event-plugins/XhrPlugin.ts:225]: agreed

File: src/plugins/event-plugins/__tests__/XhrPlugin.test.ts (hunk 1 above)
Comment (qhanam) [reply src/plugins/event-plugins/__tests__/XhrPlugin.test.ts:68]: Each test should verify one property. Doing this on every test adds unnecessary complexity.

File: src/plugins/event-plugins/XhrPlugin.ts [hunk 3]
```diff
@@ -95,13 +95,13 @@ export const XHR_PLUGIN_ID = 'xhr';
  */
 export class XhrPlugin extends MonkeyPatched<XMLHttpRequest, 'send' | 'open'> {
     private config: HttpPluginConfig;
-    private xhrMap: Map<XMLHttpRequest, XhrDetails>;
+    private map: Map<XMLHttpRequest, XhrDetails>;
```
Comment (qhanam) [inline src/plugins/event-plugins/XhrPlugin.ts:98]: Why are we doing all this renaming? This makes it difficult to understand the change. For example, if someone is debugging a related issue in the future. 

My understanding was that this would be reverted before being merged.

File: src/plugins/event-plugins/XhrPlugin.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (qhanam) [reply src/plugins/event-plugins/XhrPlugin.ts]: This is out of scope for this change -- it is unrelated to the bug fix.

"XHR" is an abbreviation of "XML HTTP Request", so "xhrRequest" is redundant.

[the remark this answers is not in the document]
Comment (williazz) [reply src/plugins/event-plugins/XhrPlugin.ts]: These renamings were to avoid repeating Xhr so many times, which is implied in plugin named XhrPlugin. I've created a separate PR #455 to restore these changes and remove the public getter, as you requested
